File: content/tracks/dsa/problems/lc-0001-two-sum/Solution.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indexOf = new HashMap<>(); // value -> index, for the numbers already passed
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (indexOf.containsKey(complement)) {
                return new int[] {indexOf.get(complement), i};
            }
            indexOf.put(nums[i], i);
        }
        return new int[0]; // unreachable: the input always has exactly one answer
    }
}

File: content/tracks/dsa/problems/lc-0001-two-sum/solution.py
from typing import Dict, List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index_of: Dict[int, int] = {}  # value -> index, for the numbers already passed
        for i, num in enumerate(nums):
            complement = target - num
            if complement in index_of:
                return [index_of[complement], i]
            index_of[num] = i
        return []  # unreachable: the input always has exactly one answer
